Sources/PenguinParallel/NonblockingThreadPool/NonBlockingThreadPool.swift

```diff
@@ -292,6 +292,40 @@ public class NonBlockingThreadPool<Environment: ConcurrencyPlatform>: ComputeThr
     if let e = err { throw e }
   }
 
+  public func parallelFor(n: Int, _ fn: VectorizedParallelForFunction) {
```

Comment:
Doc comment please!!!

Unless "parallel for on a thread pool" is a very well-established concept, I'd consider renaming these.  `for` in Swift is something we do over a `Sequence`, and there's no sequence here.  Is this something that could be written as an extension on `Collection` that accepts a thread pool as an argument?  Your `n` repetitions could be well-represented by `0..<n`.

Reply:
- Will comment on the doc comment separately (GitHub doesn't make this super convenient)
- All collections are "splittable" for reasonable definitions of the term, but maybe you mean collections whose disjoint slices can be mutated in parallel.  The more general concept is those that “have disjoint parts that can be projected for mutation in parallel.”  You could imagine a collection of pairs where you mutate the first of each pair in one thread and the second in another.
- Not quite sure what you wanted me to notice at that link.  The main thing I took away was, “why is he rebasing those slices?“ which probably wasn't the point 😉
- IMO it's questionable whether we really want the non-vectorized ones and whether they should have the same spelling anyway.